File: perform_string_shifts/0201_bitwise_and/range_bitwise_and.go
package main

/**
 * Time: O(1)
 * Space: O(1)
 */
func rangeBitwiseAnd(m int, n int) int {
	shifts := 0
	for m < n {
		m >>= 1
		n >>= 1
		shifts++
	}
	return m << uint(shifts)
}

func main() {

}

File: perform_string_shifts/0201_bitwise_and/range_bitwise_and.py
def range_bitwise_and(m: int, n: int) -> int:
    """
    Time: O(1)
    Space: O(1)
    """
    shifts = 0   
    # find the common prefix
    while m < n:
        m = m >> 1
        n = n >> 1
        shifts += 1
    return m << shifts